test/plugins/compliance_visitor_data_request_plugin_test.rb

```diff
@@ -0,0 +1,159 @@
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": visitor_email})
+        ApiResource.create(api_namespace_id: @namespace_with_email_two.id, properties: {"name": "Test Name", "email": visitor_email})
+
+        perform_enqueued_jobs do
+            @data_request_plugin.run
+            Sidekiq::Worker.drain_all
+        end
+
+        # Visitor one making submissions under two namespaces, so the email should contain two attachments
+        sent_email = ActionMailer::Base.deliveries.last
+        assert_equal 2, sent_email.attachments.length
+
+        sent_email.attachments.each do |attachment|
+            assert(attachment.content_type.start_with?("text/csv"))
+        end
+    end
+
+    test "CSV file should have the right format" do
+        visitor_email = @visitor_one.properties["email"]
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": visitor_email})
+
+        perform_enqueued_jobs do
+            @data_request_plugin.run
+            Sidekiq::Worker.drain_all
+        end
+
+        sent_email = ActionMailer::Base.deliveries.last
+        assert(sent_email.attachments[0].filename.include?("api_namespace_#{@namespace_with_email_one.name}"))
+        assert(sent_email.attachments[0].filename.include?(".csv"))
+    end
+
+    test "CSV file should contain the right content" do
+        visitor_email = @visitor_one.properties["email"]
+        api_resource = ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": visitor_email})
+
+        @namespace_with_email_one.non_primitive_properties.create([
+            {
+                label: 'richtext',
+                field_type: 'richtext'
+            }
+        ])
+
+        api_resource.non_primitive_properties.create([
+            {
+                label: 'richtext',
+                field_type: 'richtext',
+                content: "<div>Hello World</div>"
+            }
+        ])
+
+        richtext = api_resource.non_primitive_properties.find_by(label: "richtext").content.to_s
+
+        expected_csv = CSV.generate do |csv|
+            csv << ["id", "api_namespace_id", "post", "email", "created_at", "updated_at", "richtext"]
+            csv << [api_resource.id, api_resource.api_namespace_id, api_resource.properties["post"], api_resource.properties["email"], api_resource.created_at, api_resource.updated_at, richtext]
+        end
+
+        perform_enqueued_jobs do
+            @data_request_plugin.run
+            Sidekiq::Worker.drain_all
+        end
+
+        sent_email = ActionMailer::Base.deliveries.last
+        assert_equal expected_csv, sent_email.attachments.first.body.raw_source.gsub(/\r/, "")
+    end
+
+    test "Should raise an exception if visitor does not exclude any api namespace as well as does not give permission to scan all namespaces" do
```

Comment:
`visitor` `user` `current_user` and `admin` are keywords. Lets not conflate them 

rename test name to: `Should raise an exception if admin does not exclude any api namespace and does not grant explicit permission to scan all API Namespaces`